### PR TITLE
Event transitions 🎉

### PR DESCRIPTION
**Pull request by JustinStitt**
### Demo
show1

### CODE REVIEW

File: src/lib/components/events/event-item.svelte
Comment (JustinStitt) [inline src/lib/components/events/event-item.svelte]: Without this _hack_ if the user clicks the "Join" button on an event it will toggle the `shown` state causing an inverse behavior, I added this to retoggle `shown` since clicking the "Join" button counts as clicking on the Event itself.

File: src/lib/components/events/event-item.svelte
Comment (diamondburned) [inline src/lib/components/events/event-item.svelte]: This breaks without JavaScript. We should keep a `<noscript>` tag outside of the `if` block like so:

```html
<noscript>
  <p class="event-description">
    {@html info.description}
  </p>
</noscript>
{#if shown}
  <!-- same p tag as above -->
{/if}
```

**Review by diamondburned — CHANGES_REQUESTED** [review]

**Comment by diamondburned** [comment]
As a side note, the Algo event's description escapes the `<p>` tag, probably because nested `<p>` tags are invalid. It should probably be in a `<div>` instead. Note the `HTML_TAG_START` and `HTML_TAG_END` comments.

cc @EthanThatOneKid 

image

**Comment by diamondburned** [comment]
Also, can we squash these commits?

**Comment by JustinStitt** [comment]
> Also, can we squash these commits?

I like have PR revision history. Perhaps maintainer can squash pre-merge?

**Comment by JustinStitt** [comment]
Also, this [package](https://github.com/SaekiTominaga/customelements-details-animation) may be of interest. It does detail-summary animations in-house.

Adds a bit of css bloat but simplifies the DOM.

**Review by JustinStitt — COMMENTED** [review]
I have the power to LGTM but won't :)

**Comment by diamondburned** [comment]
> As a side note, the Algo event's description escapes the `<p>` tag, probably because nested `<p>` tags are invalid. It should probably be in a `<div>` instead. Note the `HTML_TAG_START` and `HTML_TAG_END` comments.

After looking into it, this doesn't happen with JS enabled. It might be the server-side rendering attempting to sanitize the HTML differently compared to the browser.

**Comment by jaasonw** [comment]
am i missing something or it seem kinda janky to me?

**Comment by JustinStitt** [comment]
> am i missing something or it seem kinda janky to me?

The behavior or my implementation?

**Comment by jaasonw** [comment]
the behavior

**Comment by jaasonw** [comment]
https://user-images.githubusercontent.com/8981287/189256563-8a2cc082-ccc6-40f3-acef-154185aa3c03.mp4
like it just kinda snaps down and when u minimize it kinda just disappears

**Comment by JustinStitt** [comment]
> https://user-images.githubusercontent.com/8981287/189256563-8a2cc082-ccc6-40f3-acef-154185aa3c03.mp4 like it just kinda snaps down and when u minimize it kinda just disappears

hmm, idk. I just tried to be clever and use the built-in Svelte transitions. It does look weird to me as well.

**Review by EthanThatOneKid — APPROVED** [review]
I appreciate the changes in this PR because @JustinStitt was able to create an enhancement that gets added to the page progressively when JavaScript is enabled. This means that he took into account backwards-compatibility, allowing the component to properly function even when the expected JavaScript code is absent.

**Review by diamondburned — APPROVED** [review]

File: src/lib/components/events/event-item.svelte
Comment (EthanThatOneKid) [reply src/lib/components/events/event-item.svelte]: This comment may be used to create a follow up issue in the future. Resolving to merge the PR.

**Comment by jaasonw** [comment]
wait hold up, am i the only one that finds this initial jerk kinda weird

https://user-images.githubusercontent.com/8981287/194151540-0bbb4174-672d-4b2e-b814-e4b3cd7819b9.mov

**Comment by JustinStitt** [comment]
I am not seeing an initial jerk. It's animation timing is probably a Cubic Bezier with a moderately high starting slope but that is just part of the animation.

This is preferred to a slow, unresponsive, and less punchy _ease-in_ timing imo.

**Comment by diamondburned** [comment]
> I am not seeing an initial jerk.

Probably because the animation timing is very low. By doing this change:

```diff
--- a/src/routes/events/event.svelte
+++ b/src/routes/events/event.svelte
@@ -93,7 +94,7 @@
     </noscript>
 
     {#if shown}
-      <p class="event-description" transition:slide>
+      <p class="event-description" transition:slide={{ duration: 2000 }}>
         {@html info.description}
       </p>
     {/if}
```

You'll see that the action buttons inside each event are not actually transitioned properly:

[Screencast from 05-10-22 15:24:01.webm](https://user-images.githubusercontent.com/8463786/194175000-de8e9d17-8ce4-4b1f-a0de-781498344284.webm)

This probably explains the initial jerk.

**Comment by jaasonw** [comment]
oh, good catch, yea these buttons needa be transitioned as well as the description
image

**Comment by jaasonw** [comment]
https://user-images.githubusercontent.com/8981287/209519742-0600c6d0-8de0-4ee4-a158-aca82df6ff70.mp4

**Comment by jaasonw** [comment]
it works but i had to do it on this older version of the website because no gcal api key

**Pull request merged**
